sycl/source/handler.cpp

```diff
@@ -499,7 +499,8 @@ void handler::associateWithHandlerCommon(detail::AccessorImplPtr AccImpl,
                                          int AccTarget) {
   detail::Requirement *Req = AccImpl.get();
   // Add accessor to the list of requirements.
-  CGData.MRequirements.push_back(Req);
+  if (Req->MAccessRange.size() != 0)
```

Comment:
Just to make sure I understand correctly, this is the only real change in this PR, right? The rest comes from #10780, right?

Reply:
Yes. One can look into the last commit in this PR to limit the diff to the actual change. I plan to mark the PR as ready once I rebase on top of origin/sycl with your PR merged.